File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        groups = collections.defaultdict(list)

        for  s in strs:
            # key = [0 for _ in range(26)]
            # for char in s:
            #     key[ord(char)-97] += 1
            
            # key = 
            key = "".join(sorted(s))
            groups[key].append(s)
        
        return list(groups.values())

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, ArrayList<String>> groups = new HashMap<>();


        for(String s:strs){
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sortedStr = new String(chars);

            if (! groups.containsKey(sortedStr)){
               groups.put(sortedStr, new ArrayList<String>());
            }
  
            groups.get(sortedStr).add(s);

        }

        return new ArrayList<>(groups.values());
    }
}